internal/weldrcheck/blueprints_test.go

```diff
@@ -0,0 +1,754 @@
+	require.True(t, resp.Status, "POST failed: %#v", resp)
+
+	// Delete the blueprint
+	resp, err = client.DeleteBlueprintV0(testState.socket, "test-delete-blueprint-v0")
+	require.NoError(t, err, "DELETE failed with a client error")
+	require.True(t, resp.Status, "DELETE failed: %#v", resp)
+}
+
+// delete a non-existent blueprint
+func TestDeleteNonBlueprint0(t *testing.T) {
+	resp, err := client.DeleteBlueprintV0(testState.socket, "test-delete-non-blueprint-v0")
+	require.NoError(t, err, "failed with a client error")
+	require.False(t, resp.Status, "did not return an error")
+}
+
+// delete a new blueprint from the workspace
+func TestDeleteNewWorkspaceV0(t *testing.T) {
+	// POST a blueprint to delete
+	bp := `{
+		"name": "test-delete-new-blueprint-ws-v0",
+		"description": "deleteNewBlueprintWSV0",
+		"version": "0.0.1",
+		"packages": [{"name": "bash", "version": "*"}],
+		"modules": [{"name": "util-linux", "version": "*"}],
+		"customizations": {"user": [{"name": "root", "password": "qweqweqwe"}]}
+	}`
+
+	resp, err := client.PostJSONWorkspaceV0(testState.socket, bp)
+	require.NoError(t, err, "POST failed with a client error")
+	require.True(t, resp.Status, "POST failed: %#v", resp)
+
+	// Delete the blueprint
+	resp, err = client.DeleteWorkspaceV0(testState.socket, "test-delete-new-blueprint-ws-v0")
+	require.NoError(t, err, "DELETE failed with a client error")
+	require.True(t, resp.Status, "DELETE failed: %#v", resp)
+}
+
+// delete blueprint changes from the workspace
+func TestDeleteChangesWorkspaceV0(t *testing.T) {
+	// POST a blueprint first
+	bp := `{
+		"name": "test-delete-blueprint-changes-ws-v0",
+		"description": "deleteBlueprintChangesWSV0",
+		"version": "0.0.1",
+		"packages": [{"name": "bash", "version": "*"}],
+		"modules": [{"name": "util-linux", "version": "*"}],
+		"customizations": {"user": [{"name": "root", "password": "qweqweqwe"}]}
+	}`
+
+	resp, err := client.PostJSONBlueprintV0(testState.socket, bp)
+	require.NoError(t, err, "POST blueprint failed with a client error")
+	require.True(t, resp.Status, "POST blueprint failed: %#v", resp)
+
+	// Post blueprint changes to the workspace
+	bp = `{
+		"name": "test-delete-blueprint-changes-ws-v0",
+		"description": "workspace copy",
+		"version": "0.2.0",
+		"packages": [{"name": "frobozz", "version": "*"}],
+		"modules": [{"name": "util-linux", "version": "*"}],
+		"customizations": {"user": [{"name": "root", "password": "qweqweqwe"}]}
+	}`
+
+	resp, err = client.PostJSONWorkspaceV0(testState.socket, bp)
+	require.NoError(t, err, "POST workspace failed with a client error")
+	require.True(t, resp.Status, "POST workspace failed: %#v", resp)
+
+	// Get the blueprint, make sure it is the modified one and that changes = true
+	info, api, err := client.GetBlueprintsInfoJSONV0(testState.socket, "test-delete-blueprint-changes-ws-v0")
+	require.NoError(t, err, "GET blueprint failed with a client error")
+	require.Nil(t, api, "GET blueprint request failed: %#v", api)
```

Comment:
Shouldn't this be `require.Nilf` or am I not understanding something ?

Reply:
I haven't figured out what the 'f' suffix is for. The docs, and in practice, indicate that all of them support formatted error messages. So I'm not sure what their purpose is.

Reply:
hmm, my impression was that -f methods supported string formatting and the others didn't. At least this is how I've used them in my limited experience. But if you say it works and I am not going to argue.